storage/sql/common/flag.go

```diff
@@ -119,20 +131,52 @@ func (s *Store) ListFlags(ctx context.Context, opts ...storage.QueryOption) ([]*
 			&flag.Enabled,
 			&createdAt,
 			&updatedAt); err != nil {
-			return nil, err
+			return results, err
 		}
 
 		flag.CreatedAt = createdAt.Timestamp
 		flag.UpdatedAt = updatedAt.Timestamp
 
 		if err := s.variants(ctx, flag); err != nil {
-			return nil, err
+			return results, err
 		}
 
 		flags = append(flags, flag)
 	}
 
-	return flags, rows.Err()
+	if err := rows.Err(); err != nil {
+		return results, err
+	}
+
+	var next *flipt.Flag
+
+	if len(flags) > int(params.Limit) && params.Limit > 0 {
+		next = flags[len(flags)-1]
+		flags = flags[:params.Limit]
+	}
+
+	results.Results = flags
+
+	if next != nil {
+		var out bytes.Buffer
+		if err := json.NewEncoder(&out).Encode(PageToken{Key: next.Key, Offset: offset + uint64(len(flags))}); err != nil {
+			return results, fmt.Errorf("encoding page token %w", err)
+		}
+		results.NextPageToken = out.String()
```

Comment:
Similar take it or leave it:
```suggestion
		out, err := json.Marshal(PageToken{Key: next.Key, Offset: offset + uint64(len(flags))})
		if err != nil {
			return results, fmt.Errorf("encoding page token %w", err)
		}
		
		results.NextPageToken = string(out)
```

Also, `json.Encoder` will newline delimit its output. Which I don't think it problematic.
Particularly, since we're base64 encoding it on the wire.
But it is superfluous.